assignments/array-methods.js

```diff
@@ -54,30 +57,53 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 {"id":50,"first_name":"Shell","last_name":"Baine","email":"sbaine1d@intel.com","shirt_size":"M","company_name":"Gabtype","donation":171}];
 
 // ==== Challenge 1: Use .forEach() ====
-// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
+// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names 
+// into a new array called fullName. 
 let fullName = [];
+runners.forEach(function(element) {
+    fullName.push(`${element.first_name} ${element.last_name}`);
+});
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
-// The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
+// The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. 
+// Convert each first name into all caps and log the result
+let allCaps = runners.map(elem => elem.first_name.toUpperCase());
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
-// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they 
+// can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of 
+// L and log the result
+let largeShirts = runners.filter(elem => elem.shirt_size === "L" );
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
-// The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+// The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the 
+// result
+let ticketPriceTotal = runners.reduce( (acc, curr) => acc + curr.donation. 0 );
```

Comment:
Check this function out again, you have some syntax issue 😄 But once you fix it, it will return the proper response.